wasm-pack.yaml

```
@@ -0,0 +1,151 @@
+      - rust
+      - zstd-dev
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/rustwasm/wasm-pack
+      expected-commit: e25150ecd322fc03c9abcd141bc979b108e8a344
+      tag: v${{package.version}}
+
+  - uses: patch
+    with:
+      patches: upgrade-deps.patch
+
+  - runs: |
+      # prepare
+      rm -rf Cargo.lock
+      cargo fetch --target="${{host.triplet.rust}}"
+
+      cargo auditable build --release --frozen
+      install -Dm755 target/release/wasm-pack \
+      -t "${{targets.contextdir}}"/usr/bin
+
+  - uses: strip
+
+update:
+  enabled: true
+  github:
+    identifier: rustwasm/wasm-pack
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - nodejs
+        - npm
+        - rustup
+        - yarn
+        - build-base
+        - binaryen
+    environment:
+      HOME: /home/build
+  pipeline:
+    - runs: |
+        # Function to display messages
+        function print_info {
+            echo -e "\033[1;34m[INFO]\033[0m $1"
+        }
+
+        function print_error {
+            echo -e "\033[1;31m[ERROR]\033[0m $1"
+            exit 1
+        }
+
+        print_info "Installing default Rust toolchain via Rustup..."
+        rustup install stable || print_error "Failed to install Rust via Rustup"
+        rustup default stable || print_error "Failed to set default Rust toolchain"
+        ARCH=$(uname -m)
+        export PATH="$HOME/.rustup/toolchains/stable-${ARCH}-unknown-linux-gnu/bin:$PATH"
+        cargo version
+
+        # Ensure the wasm32-unknown-unknown target is installed
+        print_info "Checking for wasm32-unknown-unknown target..."
+        set +e
+        rustup target list --installed | grep wasm32-unknown-unknown &> /dev/null
+        if [ $? -ne 0 ]; then
+            print_info "Installing wasm32-unknown-unknown target..."
+            rustup target add wasm32-unknown-unknown || print_error "Failed to install wasm32-unknown-unknown target."
+        else
+            print_info "wasm32-unknown-unknown target is already installed."
+        fi
+        set -e
+
+        print_info "Checking for wasm32-wasi target..."
+        rustup target list
+        # Create a new Rust library project
+        print_info "Creating a new Rust project: $PROJECT_NAME"
+        PROJECT_NAME="wasm_pack_test"
+        cargo new --lib $PROJECT_NAME || { print_error "Failed to create a new Rust project."; }
+        cd $PROJECT_NAME || { print_error "Failed to enter project directory."; }
+
+        mv ../Cargo.toml Cargo.toml
+        mv ../src/lib.rs src/lib.rs
+
+        # Clean the project to avoid any previous build issues
+        print_info "Cleaning the project..."
+        cargo clean
+
+        # Build the project with wasm-pack
+        print_info "Building the project with wasm-pack"
+        wasm-pack build --target nodejs || { print_error "Build failed. Running diagnostics..."; cargo build --target wasm32-unknown-unknown; }
```

```suggestion
        wasm-pack build --target nodejs || print_error "Build failed. Running diagnostics..."
```

`print_error` exits now, and I'm not sure what `cargo build`ing would have done anyway before.